### ISSUE TITLE
Increase the queue length and chunk size for upload using amplify.

### ISSUE THREAD

**Issue by deepak-agarwal** [aws-amplify/amplify-js#8772]
### Is this related to a new or existing framework?

React

### Is this related to a new or existing API?

Storage

### Describe the feature you'd like to request

Amplify storage uploads larger files by dividing it in a chunk of 5mb and with a concurrent upload of 4 which is hardcoded in the library. Which is difficult if you are uploading file greater than 1gb.

### Describe the solution you'd like

We can include option in Storage.put where we can pass the desired queue size and chunk size so that users can take decision based on the file size.

### Describe alternatives you've considered

We can do this using Evaporate.js , but an inbuilt feature to the library would be better.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by deepak-agarwal** [comment]
Hey @jamesaucode hope you are doing well. I can submit a PR for this but not able to create an issue here.

**Comment by abdallahshaban557** [comment]
Hi @deepak-agarwal  - We appreciate you sharing this suggestion with us! Our team will explore enabling this in the future!

**Comment by dbanksdesign** [comment]
Android and iOS libraries calculate the optimal chunk size based on the file:

https://github.com/aws-amplify/amplify-android/blob/2f71e3e6f8a1d53c2e22c6b34b3c9d7ffa5abd75/aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/TransferManager.kt#L231

https://github.com/aws-amplify/amplify-swift/blob/7f094a9e88612b5b8fda1bce01ce71e7f1915053/AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/StorageUploadPart.swift#L152

**Comment by adi518** [comment]
This is a really bad bug. Any workarounds?

**Comment by AllanZhengYP** [comment]
While working on a fix to this issue, I found another issue with the upload implementation. The uploader read the file into the memory and split the data into all chunks before starting to upload:
https://github.com/aws-amplify/amplify-js/blob/0fa40e6efb67def43f0eb912b3f2ed34da883f5f/packages/storage/src/providers/AWSS3ProviderManagedUpload.ts#L84-L96

This limits the upload object size to the browser memory limit(i.e. 16 GB for Chromium). For bigger objects, the process would be failed memory/array allocation error.

**Comment by adi518** [comment]
I ended up working around it with `patch-package` and simply shoved a condition in there, to alternate between the default file part size and a large part size for large files, to keep the number of parts well under the limit of the S3 spec.

**Comment by AllanZhengYP** [comment]
@adi518 Thank you for the update. I'm testing locally on a machine with 16GB memory. Maybe I'm hitting the physical memory limit not the browser limit. Can you share what machine your test is conducted on?

**Comment by adi518** [comment]
M1 MPB 16", 32GB of RAM.

**Comment by AllanZhengYP** [comment]
Hi @deepak-agarwal @adi518 ,

I pushed my fix to a tagged release `storage-part-size`. Can you install this package `yarn add aws-amplify@storage-part-size` so see if the issue has been fixed?

**Comment by abdallahshaban557** [comment]
Closing this issue as resolved by #10820!
